lib/pbio/drv/legodev/legodev_pup_uart.c

```diff
@@ -726,6 +729,9 @@ static PT_THREAD(pbdrv_legodev_pup_uart_update(pbdrv_legodev_pup_uart_dev_t * da
     uint8_t speed_payload[4];
     pbio_set_uint32_le(speed_payload, EV3_UART_SPEED_LPF2);
     ev3_uart_prepare_tx_msg(data, LUMP_MSG_TYPE_CMD, LUMP_CMD_SPEED, speed_payload, sizeof(speed_payload));
+
+    pbdrv_uart_flush(data->uart);
```

Comment:
Why do we need the flush here since we are already flushing on line 741?

Reply:
Without this, it always syncs up at 2400 baud.

Reply:
I still don't understand how this could make a difference since nothing is read between this flush and the next flush.

Reply:
That makes two of us :smile:

Reply:
I did notice that on boot, it does always work on high baud the first time on Prime Hub, but on plugging in the second time, it usually goes at `2400` baud, as if something (else) isn't quite flushed out.

The easiest way to see is with the SPIKE Color Sensor, whose light will either turn on nearly instantaneously or otherwise take a few seconds to dump all its data.